tensornetwork/block_sparse/blocksparsetensor.py

```diff
@@ -626,13 +626,33 @@ def _sub_add_protection(self, other):
       raise ValueError(
           "cannot add or subtract tensors non-matching sparse shapes")
 
+  def _align_storage_layout(self, other):
+    """
+    Align storage layouts of self and other.
+    """
+    self_is_ordered = np.array_equal(self.flat_order,
+                                     np.arange(len(self.flat_order)))
+    other_is_ordered = np.array_equal(other.flat_order,
+                                      np.arange(len(other.flat_order)))
+    if self_is_ordered and (not other_is_ordered):
+      #bring other into the same storage layout as other
+      perm = np.empty(len(other.flat_order), dtype=np.int32)
+      perm[self.flat_order] = other.flat_order
+      other.transpose_data(perm, inplace=True)
+    elif (not self_is_ordered) and other_is_ordered:
```

Comment:
This and the `else` below appear to be the same code. Can you just make it a single `else`?

Reply:
They are, I thought it might be easier to parse for the reader this way, this is why I split it up

Reply:
I'll fix it!